.github/workflows/deploy.yml

```diff
@@ -102,7 +102,20 @@ jobs:
         name: Upload files to S3 bucket
         if: ${{ env.DOCS_S3_BUCKET != '' }}
         run: |
-          aws --region ${{ env.DOCS_AWS_REGION }} s3 sync --acl public-read public s3://${{ env.DOCS_S3_BUCKET }}/ --delete
+          aws --region ${{ env.DOCS_AWS_REGION }} s3 sync \
+            --acl public-read \
+            --delete \
```

Comment:
per the documentation, the `--delete` should respect `--exclude`
https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html

> `--delete` (boolean) Files that exist in the destination but not in the source are deleted during sync. Note that files excluded by filters are excluded from deletion.